### PR TITLE
[opt](nereids) Join related cost calculation formula works well at least 3be num

### PR DESCRIPTION
**Pull request by englefly**
## Proposed changes

In previouse #37823, we adjust the shuffle and join cost. However the new cost model is not good to single BE.
the  new cost model count BE number at least 3

Issue Number: close #xxx

### CODE REVIEW

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run cloud_p1

File: fe/fe-core/src/main/java/org/apache/doris/nereids/cost/CostModelV1.java [hunk 1]
```diff
@@ -271,14 +271,14 @@ public Cost visitPhysicalDistribute(
         Statistics childStatistics = context.getChildStatistics(0);
         double intputRowCount = childStatistics.getRowCount();
         DistributionSpec spec = distribute.getDistributionSpec();
-
+        int beNumForDist = Math.max(3, beNumber);
```
Comment (xzj7019) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/cost/CostModelV1.java:274]: pls add comments here

Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/cost/CostModelV1.java:274]: let this local var be a CostModelV1's member var?

Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/cost/CostModelV1.java:275]: done

Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/cost/CostModelV1.java:275]: this is used for distribute and join. Other SQL operator still use beNumber. I think it is better to make it function scope

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run p0

**Comment by englefly** [comment]
run cloud_p1

**Review by xzj7019 — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**